lib/posix/posix_utils.nim

```diff
@@ -127,3 +127,21 @@ proc osReleaseFile*(): Config {.since: (1, 5).} =
     if fileExists(osReleaseFile):
       return loadConfig(osReleaseFile)
   raise newException(IOError, "File not found: /etc/os-release, /usr/lib/os-release")
+
+proc setHostName*(name: string) {.raises: [OSError], sideEffect.} =
+  ## Sets host name. Available in Linux (including Android), OS X, BSD, Haiku,
+  ## and QNX distributions. Raises `OSError` on other POSIX platforms.
+  var callResult: cint
+  when (defined(android) or defined(haiku) or defined(linux) or
+        defined(netbsd) or defined(openbsd) or defined(qnx)):
+    proc sethostname(name: cstring, len: csize_t): cint
+      {.importc, header: "<unistd.h>", sideEffect.}
+    callResult = sethostname(name.cstring, csize_t(name.len))
+  elif defined(dragonfly) or defined(freebsd) or defined(osx):
+    proc sethostname(name: cstring, len: cint): cint
+      {.importc, header: "<unistd.h>", sideEffect.}
+    callResult = sethostname(name.cstring, cint(name.len))
+  else:
+    raise newException(OSError, "setHostName failed: OS is not supported")
+  if callResult != 0:
+    raise newException(OSError, $strerror(errno))
```

Comment:
in addition, always show relevant runtime context (name) in exception, see how it's done in os.nim